### PR TITLE
[JIMT]all-songs-available

### PR DESCRIPTION
**Pull request by thomasoniii**
This is an updated to #54208 

Three small things -
• When no songs are selected, then all songs are available. Change the output on the "Songs currently" selected paragraph to say "All songs are available" instead of `[]`
• If no default song was selected, then it says "No song chosen"  
• There was a bug creating a new level because it was assuming a default song was already populated (testing miss). So just shoves in a default value.

Screenshot 2023-10-18 at 2 12 16 PM

### CODE REVIEW

File: dashboard/app/views/levels/editors/fields/_default_song.html.haml [hunk 1]
```diff
@@ -22,4 +22,4 @@
 
 %p Previously chosen default song (may not match actual default if song selection was limited):
 
-= songMap[@level.default_song][0]
+= (songMap[@level.default_song]||[])[0]
```
Comment (thomasoniii) [inline dashboard/app/views/levels/editors/fields/_default_song.html.haml]: This is ugly syntax. Does ruby have a slicker construct? Something like an optional chaining operator?

**Review by fisher-alice — COMMENTED** [review]
Could you please add a couple screenshots of the updates and a link to your original PR? Thank you!

**Comment by thomasoniii** [comment]
> Could you please add a couple screenshots of the updates and a link to your original PR? Thank you!

Added!

**Review by jmkulwik — APPROVED** [review]
LGTM!

**Comment by fisher-alice** [comment]
> > Could you please add a couple screenshots of the updates and a link to your original PR? Thank you!
> 
> Added!

Thanks for the quick fixes and updates!

**Pull request merged**